### ISSUE TITLE
[BUG] `patterns may only match literals and attribute lookups` with mix-str

### ISSUE THREAD

**Issue by hugocarnide** [dashingsoft/pyarmor#1358]
Please read this first before report any issue
https://pyarmor.readthedocs.io/en/latest/questions.html

When using the python structural pattern matching and option mix-str from pyarmor, pyarmor cannot process the obfuscation.
When the pattern in the case have more than 8 characters for a string it is refused. The script work as it.
If in the sample, the character 9 is remove, the pyarmor can process until the end.

Here the source code sample
```python 
import sys
pattern_match = "create"
if len(sys.argv) == 2:
    pattern_match = sys.argv[1]
match pattern_match:
    case "create":
        print(1)
    case "123456789":
        print(2)
    case _:
        raise ValueError(f"Unknown command")
```

Please copy the whole command line `pyarmor gen` and first 4 lines in the console, do not mask anything, and do not paste snapshot image

```
pyarmor gen cli.py --mix-str
INFO     Python 3.11.2
INFO     Pyarmor 8.2.9 (basic), 003609, C. Softwares
INFO     Platform windows.x86_64
INFO     search inputs ...
INFO     find script cli.py
INFO     find 1 top resources
INFO     start to generate runtime files
INFO     target platforms {'windows.amd64'}
INFO     write dist\pyarmor_runtime_003609\pyarmor_runtime.pyd
INFO     patch runtime file
INFO     generate runtime files OK
INFO     start to obfuscate scripts
INFO     process resource "cli"
INFO     obfuscating file cli.py
ERROR    patterns may only match literals and attribute lookups
```

**Comment by jondy** [comment]
Reproduced, it will be fixed in next release.

**Comment by jondy** [comment]
Fixed in 8.3.dev1
